src/components/TimePicker.tsx

```diff
@@ -0,0 +1,81 @@
+import './styles/time_picker.less'
+
+import React from 'react'
+import ClockIcon from './icons/ClockIcon.svg'
+
+import CustomParseFormat from 'dayjs/plugin/customParseFormat'
+import dayjs, { Dayjs } from 'dayjs'
+import Select from './select'
+import classNames from 'classnames'
+
+dayjs.extend(CustomParseFormat)
+
+const MINUTES_INTERVAL = 5
+
+type Props = {
+    minutesInterval?: number
+    onTimeChanged?: (value: string) => void
+    min?: string
+    max?: string
+    defaultValue?: string
+    className?: string
+    optionsClassName?: string
+}
+
+export default function TimePicker({
+    minutesInterval,
+    onTimeChanged,
+    min,
+    max,
+    defaultValue,
+    className,
+    optionsClassName,
+}: Props): JSX.Element {
+    const times = [
+        {
+            value: '',
+            text: '--:--',
```

Comment:
we could stick to track the value only, and generate the text from the value

Reply:
By doing it this way, it meant I could just drop the times into the `options` property on the `Select` without having to do any additional work.